src/codegen.c

```diff
@@ -0,0 +1,591 @@
+write_veci(f, M->p, M->n+1, vec_name);
+sprintf(vec_name, "%s_i", name);
+write_veci(f, M->i, M->nzmax, vec_name);
+sprintf(vec_name, "%s_x", name);
+write_vecf(f, M->x, M->nzmax, vec_name);
+fprintf(f, "csc %s = {\n", name);
+fprintf(f, "  %d,\n", M->m);
+fprintf(f, "  %d,\n", M->n);
+fprintf(f, "  %s_p,\n", name);
+fprintf(f, "  %s_i,\n", name);
+fprintf(f, "  %s_x,\n", name);
+fprintf(f, "  %d,\n", M->nzmax);
+fprintf(f, "  %d,\n", M->nz);
+fprintf(f, "};\n");
+}
+
+static void write_OSQPMatrix(FILE             *f,
+                             const OSQPMatrix *mat,
+                             const char       *name){
+
+  char csc_name[50];
+  sprintf(csc_name, "%s_csc", name);
+  write_csc(f, mat->csc, csc_name);
+  fprintf(f, "OSQPMatrix %s = {\n", name);
+  fprintf(f, "  &%s,\n", csc_name);
+  fprintf(f, "  %d\n", mat->symmetry);
+  fprintf(f, "};\n");
+  
+}
+
+
+/**********
+* Settings
+***********/
+
+static void write_settings(FILE               *f,
+                           const OSQPSettings *settings,
+                           const char         *prefix){
+  
+  fprintf(f, "/* Define the settings structure */\n");
+  fprintf(f, "OSQPSettings %ssettings = {\n", prefix);
+  fprintf(f, "  0,\n"); // device
+  fprintf(f, "  OSQP_LINSYS_SOLVER,\n");
+  fprintf(f, "  0,\n"); // verbose
+  fprintf(f, "  %d,\n", settings->warm_starting);
+  fprintf(f, "  %d,\n", settings->scaling);
+  fprintf(f, "  0,\n"); // polishing
+  fprintf(f, "  (c_float)%.20f,\n", settings->rho);
+  fprintf(f, "  %d,\n", settings->rho_is_vec);
+  fprintf(f, "  (c_float)%.20f,\n", settings->sigma);
+  fprintf(f, "  (c_float)%.20f,\n", settings->alpha);
+  fprintf(f, "  %d,\n", settings->cg_max_iter);
+  fprintf(f, "  %d,\n", settings->cg_tol_reduction);
+  fprintf(f, "  (c_float)%.20f,\n", settings->cg_tol_fraction);
+  fprintf(f, "  %d,\n", settings->adaptive_rho);
+  fprintf(f, "  %d,\n", settings->adaptive_rho_interval);
+  fprintf(f, "  (c_float)%.20f,\n", settings->adaptive_rho_fraction);
+  fprintf(f, "  (c_float)%.20f,\n", settings->adaptive_rho_tolerance);
+  fprintf(f, "  %d,\n", settings->max_iter);
+  fprintf(f, "  (c_float)%.20f,\n", settings->eps_abs);
+  fprintf(f, "  (c_float)%.20f,\n", settings->eps_rel);
+  fprintf(f, "  (c_float)%.20f,\n", settings->eps_prim_inf);
+  fprintf(f, "  (c_float)%.20f,\n", settings->eps_dual_inf);
+  fprintf(f, "  %d,\n", settings->scaled_termination);
+  fprintf(f, "  %d,\n", settings->check_termination);
+  fprintf(f, "  (c_float)%.20f,\n", settings->time_limit);
+  fprintf(f, "  (c_float)%.20f,\n", settings->delta);
+  fprintf(f, "  %d,\n", settings->polish_refine_iter);
+  fprintf(f, "};\n\n");
+}
+
+
+/******
+* Info
+*******/
+
+static void write_info(FILE           *f,
+                       const OSQPInfo *info,
+                       const char     *prefix){
+  
+  fprintf(f, "/* Define the info structure */\n");
+  fprintf(f, "OSQPInfo %sinfo = {\n", prefix);
+  fprintf(f, "  \"%s\",\n", info->status);
```

Comment:
Does it make sense to populate these with non-default values as its starting values? This is the solver return info, so it would seem better to start it out as being default values (e.g. use `OSQP_UNSOLVED`).